src/software/thunderscope/gl/layers/gl_simulator_layer.py

```diff
@@ -32,22 +30,17 @@ def __init__(self, name: str, friendly_colour_yellow: bool, buffer_size: int = 5
         self.friendly_colour_yellow = friendly_colour_yellow
         self.simulator_state_buffer = ThreadSafeBuffer(buffer_size, SimulatorState)
 
-        self.graphics_list.register_graphics_group(
-            "ball",
-            lambda: GLSphere(
-                radius=BALL_MAX_RADIUS_METERS, color=Colors.SIM_BALL_COLOR
-            ),
+        self.ball_graphic = GLSphere(
+            parentItem=self, radius=BALL_MAX_RADIUS_METERS, color=Colors.SIM_BALL_COLOR
         )
 
-    def _update_graphics(self):
-        """Fetch and update graphics for the layer"""
+    def refresh_graphics(self):
```

Comment:
Also, simulator layer doesn't seem to be showing anything for me (or maybe it's drawing it somewhere incorrect?!). I tried turning off the vision layer and still couldn't see it.

Reply:
Just tried out simulator layer with vision off and it works for me, maybe try again?